### DISCUSSION TITLE
Test-Connection

### DISCUSSION THREAD

**Discussion by jNizM** [Romanitho/Winget-AutoUpdate#394] [Q&A]
May I ask why you use "Invoke-WebRequest" and not "Test-Connection"?

`Test-Connection -ComputerName www.github.com -Count 1 -Quiet`
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/test-connection?view=powershell-5.1

**Comment by jNizM** [comment]
There are some Windows 10 Clients with errors:

```
+ ... content = ((Invoke-WebRequest -URI $URLtoTest -UseBasicParsing).conte ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

**Reply by Romanitho** [reply]
We used to use Test-Connection in the past (before v1.15), but behind a captive portal the test was `True`. We had bad experience with it. But if you have a better version than `Invoke-WebRequest` that could works in all cases, I'm ok :)

**Reply by jNizM** [reply]
Maybe with Invoke-RestMethod
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-5.1

```
Invoke-RestMethod "https://raw.githubusercontent.com/Romanitho/Winget-AutoUpdate/main/LICENSE"
```

Someone with a captive portal will have to test it, of course.

**Reply by Romanitho** [reply]
not sure it will work with restmethod

**Reply by jNizM** [reply]
At least this command executed directly in Powershell returns the complete content of the LICENSE file

**Reply by Romanitho** [reply]
That's true.
